### PR TITLE
added Code libraries and OpenCage geocoder

### PR DESCRIPTION
**Pull request by freyfogle**
added provider agnostic code libraries, alpha ordering

### CODE REVIEW

File: _posts/0100-01-01-geocoding.md [hunk 1]
```diff
@@ -19,18 +19,20 @@ The data required to run a worldwide geocoder is significant and not readily ava
 
 There are many geocoders available, with varying levels of performance, coverage areas, and granularity.
 
-### Worldwide Services
+### Worldwide Services
```
Comment (tmcw) [inline _posts/0100-01-01-geocoding.md:22]: Hm, please trim this trailing space, all these newfangled editors add too much whitespace noise :)

**Comment by tmcw** [comment]
Cool - though I'm not sure about the Code Libraries part. The linked libraries aren't actually geocoders, but web service clients, and not particularly necessary ones, when geocoders tend to talk simple JSON/GeoJSON/REST

**Comment by sabas** [comment]
Yes,  isn't that section perhaps too specific for Mapschool ?

**Comment by tmcw** [comment]
Possibly a great fit for mapmakers-cheatsheet though: https://github.com/tmcw/mapmakers-cheatsheet

**Comment by tmcw** [comment]
Ported to mapmakers-cheatsheet in https://github.com/tmcw/mapmakers-cheatsheet/commit/37385dd5000558401ce2d6dd4691e082503356fd

**Pull request closed without merging**